### PR TITLE
Ignore contents of img html tags + other bits

### PR DESCRIPTION
**Pull request by GRyall**
- Ignores img html tags
- minor changes to readme and other bits
- Changes behaviour with incomplete pairs of tags and errors, rather than ignoring the whole page
- reduces allowed word list in size (no new words, check still passes)

### CODE REVIEW

File: tests/page_tests.py [hunk 1]
```diff
@@ -89,21 +88,22 @@ def strip_inline_code_blocks(text):
             expression = r"(?:(?<!\\)((?:\\{2})+)(?=`+)|(?<!\\)(`+)(.+?)(?<!`)\2(?!`))"
             return re.sub(expression, "", text)
 
+        def strip_img_html_tags(text):
+            expression = r"(<img )([a-zA-Z=\"0-9\/\-\s\.:]+)(>)"
```
Comment (Tom-Willemsen) [inline tests/page_tests.py:94]: [ZA̡͊͠͝LGΌ ISͮ̂҉̯͈͕̹̘̱ TO͇̹̺ͅƝ̴ȳ̳ TH̘Ë͖́̉ ͠P̯͍̭O̚​N̐Y̡ H̸̡̪̯ͨ͊̽̅̾̎Ȩ̬̩̾͛ͪ̈́̀́͘ ̶̧̨̱̹̭̯ͧ̾ͬC̷̙̲̝͖ͭ̏ͥͮ͟Oͮ͏̮̪̝͍M̲̖͊̒ͪͩͬ̚̚͜Ȇ̴̟̟͙̞ͩ͌͝S̨̥̫͎̭ͯ̿̔̀ͅ](https://stackoverflow.com/a/1732454)

(in this case, it's _probably_ fine...)

Comment (GRyall) [reply tests/page_tests.py:94]: could just allow all characters, but it felt like that might hit some edge cases where it matches things it shouldn't. Maybe I was worrying too much about that? That said - the case we are looking for is images added in GitHub wiki pages, usually using the editor, so we can be fairly certain there are no odd characters in there.

Comment (GRyall) [reply tests/page_tests.py:94]: Can change to `expression = r"()"` if preferred

Comment (Tom-Willemsen) [reply tests/page_tests.py:94]: To be clear, I don't actually really mind too much in this particular _limited_ case...

File: tests/page_tests.py [hunk 2]
```diff
@@ -21,15 +21,14 @@
 WIKI_INCLUDELIST = [USER_MANUAL, IBEX_MANUAL, DEV_MANUAL, TEST_WIKI]
 
 
-def strip_between_tags(expression, text):
+def strip_between_tags(self, expression, text):
```
Comment (KathrynBaker) [inline tests/page_tests.py]: This is a free function, what exactly is self here as it isn't a class?

File: tests/page_tests.py [hunk 3]
```diff
@@ -39,16 +38,16 @@ def strip_between_tags(expression, text):
 
 
 class PageTests(unittest.TestCase):
-    def __init__(self, methodName, ignored_items, wiki_info=None):
+    def __init__(self, method_name, ignored_items, wiki_info: tuple[str, list[str], str, int]):
```
Comment (KathrynBaker) [inline tests/page_tests.py]: The use of the name that the linter doesn't like is because that is the name of the method to be overloaded. See https://docs.python.org/3/library/unittest.html, worth adding a comment in the code to highlight this

File: tests/page_tests.py [hunk 4]
```diff
@@ -72,13 +71,13 @@ def replace_selected_specials_with_whitespace(text):
             return altered_text
 
         def strip_pre_tag_blocks(text):
-            return strip_between_tags(r"<pre>|</pre>", text)
+            return strip_between_tags(self, r"<pre>|</pre>", text)
```
Comment (KathrynBaker) [inline tests/page_tests.py]: If these are class functions self should be in the arguments, as at a quick glance I can't see where self comes from, or even know which self it refers to

**Comment by GRyall** [comment]
Updated in response to review and ready to merge
